src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
```

Comment:
In your `setUp` method, you're creating new instances of `RegistrationServiceImpl` and `StorageDaoImpl` before each test. This is not necessary and can lead to problems. For example, if one test modifies the state of these objects, it could affect the outcome of other tests. Instead, initialize these objects once and reuse them in all tests. You can do this by using the `@BeforeAll` annotation instead of `@BeforeEach`.